### PR TITLE
Add UFO Evaluation components for conventional surface observations

### PR DESCRIPTION
**Pull request by emilyhcliu**
- Add two YAML files for conventional surface observations: one for HofX test and the other one for QC filtering test
- Benchmarks are added to the tests
- Update run script to handle both tests
- New test data which combines the data from SFCSHIP, SFC, and ADPUPA (psob from the lowest level) is added to test directory
- A corresponding UFO changes can be found in [UFO PR #3034](https://github.com/JCSDA-internal/ufo/pull/3034)
- The conventional surface obs QC filtering included in this PR 100% reproduces GSI results (see [documentation](https://docs.google.com/presentation/d/1OvVMYOutxVyk0GE93Jzoh3hQVoj-dCgrwsVWDDlPT1s/edit?usp=sharing))

### CODE REVIEW

File: CMakeLists.txt
Comment (CoryMartin-NOAA) [inline CMakeLists.txt]: Note to revert this change before we merge this to GDASApp develop, dependent on https://github.com/JCSDA-internal/ufo/pull/3034

File: ush/ufoeval/run_ufo_hofx_test.sh
Comment (CoryMartin-NOAA) [inline ush/ufoeval/run_ufo_hofx_test.sh]: Will this path work for all ob types or just Ps?

Comment (emilyhcliu) [reply ush/ufoeval/run_ufo_hofx_test.sh]: Ha!  Fixed.  Thanks.

**Review by CoryMartin-NOAA — COMMENTED** [review]
Thanks @emilyhcliu everything looks good. Once the UFO PR is merged, we revert the CMakeLists.txt change, and I will approve and merge this.

**Review by ADCollard — DISMISSED** [review]

**Review by azadeh-gh — DISMISSED** [review]

File: CMakeLists.txt
Comment (emilyhcliu) [reply CMakeLists.txt]: @CoryMartin-NOAA I fixed the sfcpcorrected ctest in UFO [PR#3034](https://github.com/JCSDA-internal/ufo/pull/3034)
The test passed in my ufo-bundle.  The GitHub check seems to have a system issue.  The sfcpcorrected ctest should be passed once the GitHub check returns to normal.  The PR has two approvals already; it should be merged soon.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**
